io/io/src/TMemFile.cxx

```diff
@@ -134,13 +134,22 @@ TMemFile::TMemFile(const char *path, ExternalDataPtr_t data) :
    Init(/* create */ false);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor to create a TMemFile re-using external storage.
+
+TMemFile::TMemFile(const char *path, ExternalDataPtr_t data)
```

Comment:
I think this is worse than using `char*, size_t`, as the type information is unnecessarily hidden and one has to look it up to be able to call the constructor.  I also do not see why you need this and cannot use one of the other constructors (e.g. the one that takes a `std::unique_ptr<TBufferFile>`. If you still want to add this constructor, I think that receiving a `std::unique_ptr<char>` and the size is better than having the opaque struct.

Reply:
We have a blob of memory which is mmap-ed by the modules system. I need a non-owning way to form a `TMemFile`. Please take a look at the test case which I believe demonstrate the idea. If you know how to use `unique_ptr` or anything existing (which does not copy memory) for wrapping this [line](https://github.com/root-project/root/pull/3965/files#diff-d3731455c1ae88756f80d3d7001c94b8R70) into a TMemFile please let me know.